Add values for config map settings, including access log.

costinm opened the pull request:
Allow helm to configure few more istio config settings at install time.

istio-testing commented:
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/istio/istio/pull/6797#" title="Author self-approved">costinm</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[install/OWNERS](https://github.com/istio/istio/blob/master/install/OWNERS)~~ [costinm]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":[]} -->

istio-testing commented:
@costinm: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-unit-tests.sh | 06c0845d97a78297d094c358498296d4272a69dc | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/6797/istio-unit-tests/9726/) | `/test istio-unit-tests`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

rshriram commented on install/kubernetes/helm/istio/templates/configmap.yaml:
```diff
@@ -12,14 +12,17 @@ data:
     # To disable the mixer completely (including metrics), comment out
     # the following lines
+    # Deprecated: mixer is using EDS
     mixerCheckServer: istio-policy.{{ .Release.Namespace }}.svc.cluster.local:15004
```
I think people wanted to configure this as well.. especially the svc.cluster.local as some ebay folks said that they have non default domains. so having this will break their system. In fact, Pilot also has an option to specify the cluster domain (I think). That needs to percolate

costinm replied:
You're right, still needed for that. Will remove the deprecated.

Maybe we should have an option to override the entire name ( so it can be pointed to DNS entry, for multicluster ) ?

rshriram commented on install/kubernetes/helm/istio/templates/configmap.yaml:
```diff
@@ -30,6 +33,7 @@ data:
     # utilization and potential performance loss in exchange for faster
     # convergence. Tweak this value according to your setup.
+    # Deprecated: using push.
     rdsRefreshDelay: {{ .Values.global.refreshInterval }}
```
please remove.. We are not supporting any form of v1 in 1.0. So, having this here merely adds to confusion.

costinm replied:
Done.

rshriram commented on install/kubernetes/helm/istio/templates/configmap.yaml:
```diff
@@ -44,6 +48,7 @@ data:
       # the same changes in start up args in istio-ingress pods.
       # See rdsRefreshDelay for explanation about this setting.
+      # Deprecated: using push.
       discoveryRefreshDelay: {{ .Values.global.refreshInterval }}
```
same as above

sdake commented on install/kubernetes/helm/istio/templates/configmap.yaml:
```diff
@@ -12,14 +12,17 @@ data:
+
     # Set accessLogFile to empty string to disable access log.
-    accessLogFile: "/dev/stdout"
+    accessLogFile: "{{ .Values.global.proxy.accessLogFile }}"
```
+2!

sdake commented on install/kubernetes/helm/istio/values.yaml:
```diff
@@ -8,19 +8,22 @@ global:
-  # Should track latest released version in the branch.
-  tag: 0.8.latest
+  tag: nightly-master
+  #tag: 1.0.latest
```
is this leftover?

costinm replied:
No, in the branch it'll have a different value. We want each branch to default to latest nightly build ( or daily - if they start getting stable tags ). This way someone can use the templates and test the release - without setting a tag override or building.

sdake commented on install/kubernetes/helm/istio/values.yaml:
```diff
@@ -8,19 +8,22 @@ global:
-  # docker.io/istionightly
-  hub: docker.io/istio
+  hub: docker.io/istionightly
+  #hub: docker.io/istio
```
is this leftover?

costinm replied:
Same.. I can remove the commented out - and have each branch have it's own version.

sdake commented on install/kubernetes/helm/istio/values.yaml:
```diff
@@ -37,7 +40,9 @@ global:
-    policy: enabled
+
+    # This controls the 'policy' in the sidecar injector.
+    autoinject: enabled
```
should this be autoInject for consistency?

costinm replied:
good point. Policy was very confusing - we also have the mixer policy.

istio-testing commented:
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/istio/istio/pull/6797#" title="Author self-approved">costinm</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[install/OWNERS](https://github.com/istio/istio/blob/master/install/OWNERS)~~ [costinm]
- ~~[pilot/OWNERS](https://github.com/istio/istio/blob/master/pilot/OWNERS)~~ [costinm]
- ~~[tests/OWNERS](https://github.com/istio/istio/blob/master/tests/OWNERS)~~ [costinm]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":[]} -->

istio-testing commented:
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/istio/istio/pull/6797#" title="Author self-approved">costinm</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[install/OWNERS](https://github.com/istio/istio/blob/master/install/OWNERS)~~ [costinm]
- ~~[tests/OWNERS](https://github.com/istio/istio/blob/master/tests/OWNERS)~~ [costinm]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":[]} -->

costinm commented:
Sorry, messed up the branches - need to submit the long-running test changes first...

costinm commented:
Please take another look. I messed up the branches - had 2 PRs depending on each other ( the test templates ). I fixed the comments and added the matching changes for remote.

I'll do a separate PR to match remote values with the main cluster values.

istio-testing commented:
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/istio/istio/pull/6797#" title="Author self-approved">costinm</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[install/OWNERS](https://github.com/istio/istio/blob/master/install/OWNERS)~~ [costinm]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":[]} -->
